tools/Import_Data_Normalise_Consent.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * This script is written for a one time use only to normalise the consent feature of the
+ * candidate parameters module and populate the normalised tables with existing
+ * data from the participant_status and consent_info_history tables in the database.
+ * 
+ * The old tables and columns should be deleted after running this script. Follow the prompts at the end.
+ * 
+ * affected tables:
+ * - participant_status
+ * - consent_info_history
+ * - consent_type
+ * - candidate_consent_type_rel
+ * - candidate_consent_type_history
+ * 
+ * PHP Version 7
+ * 
+ * @category Main 
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+
+$config = NDB_Config::singleton();
+$db     =& Database::singleton();
+
+// Get consent details from Config.xml
+$consentConfig = $config->getSettingFromXML('ConsentModule');
+$useConsent = $consentConfig['useConsent'];
+$consents = $consentConfig['Consent'];
+
+// Check and validate existing consent details
+$errors = [];
+// Check for when consent columns exist in table but not in list of consents in Config.xml
+$columnQuery = 'SELECT Column_name FROM Information_schema.columns WHERE
+                Column_name LIKE "%consent%" AND Table_name LIKE "participant_status"';
+$existingColumns = $db->pselect($columnQuery, array());
+
+// Format result to remove columns with 'date' and 'withdrawal'
+$formattedColumns = [];
+foreach ($existingColumns as $key=>$column) {
+    $columnName = $column['Column_name']; 
+    if (!(preg_match("/date/", $columnName) || preg_match("/withdrawal/", $columnName))) {
+        array_push($formattedColumns, $columnName);
+    }
+}
+// Check that column names are in list of consent names
+foreach ($formattedColumns as $columnName) {
+    $i=0;
+    foreach ($consents as $consent){
+        $consentName = $consent['name'];
+        if($consentName===$columnName) {
+            $i++;
+        }
+    }
+    if ($i===0) {
+        array_push(
+            $errors, "The consent type " . $columnName . " exists in the database but not in Config.xml.
+           Please add the consent to Config.xml or delete columns and data from 'participant_status'"
+        );
+    }
+}
+foreach ($consents as $key=>$consent) {
+    // Do consent columns exist in old table?
+    $consentName = $consent['name'];
+    $columnQuery = 'SHOW COLUMNS FROM participant_status LIKE "' . $consentName . '"';
```

Comment:
@ridz1208 at the moment, only the columns with status information is being checked for existence in participant_status. do you think this is enough or should we also be checking for the existence of the _date and _withdrawal columns too?

Reply:
the check shouldbe done on all columns since you dont want to assume that they exist... just to be sure.